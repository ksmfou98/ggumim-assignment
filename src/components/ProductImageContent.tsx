import useProductImageSize from "hooks/useProductImageSize";
import React, { useState } from "react";
import styled from "styled-components";
import { IProduct } from "types/product";
import MagnifierList from "./MagnifierList";
import SubImageList from "./SubImageList";

interface IProductImageContentProps {
  products: IProduct;
}

export type SelectedImageIdType = number | null;

function ProductImageContent({ products }: IProductImageContentProps) {
  const { imageUrl, productList } = products;
  const [selectedImageId, setSelectedImageId] =
    useState<SelectedImageIdType>(null);

  const onSelectImage = (id: SelectedImageIdType) => setSelectedImageId(id);

  const { imageRef, adjustedImageSize, originalImageSize, onImageLoad } =
    useProductImageSize();

  return (
    <Container>
      <MainImageBlock>
        <MainImage
          ref={imageRef}
          src={imageUrl}
          alt="mainImage"
          onClick={() => onSelectImage(null)}
          onLoad={onImageLoad}
        />
        <MagnifierList
          productList={productList}
          onSelectImage={onSelectImage}
          selectedImageId={selectedImageId}
          adjustedImageSize={adjustedImageSize}
          originalImageSize={originalImageSize}
        />
      </MainImageBlock>
      <SubImageList
        productList={productList}
        onSelectImage={onSelectImage}
        selectedImageId={selectedImageId}
      />
    </Container>
  );
}

const Container = styled.div``;

const MainImageBlock = styled.div`
  position: relative;
`;

const MainImage = styled.img`
  width: 100%;
`;

export default ProductImageContent;
