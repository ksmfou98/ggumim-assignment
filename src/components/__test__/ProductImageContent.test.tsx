import { render } from "@testing-library/react";
import ProductImageContent from "components/ProductImageContent";

describe("ProductImageContent", () => {
  const initialProps = {
    products: {
      id: 89776,
      imageUrl:
        "//cdn.ggumim.co.kr/cache/star/1000/2022011017094316oRcWeb8R.jpeg",
      productList: [
        {
          productId: 219762,
          productName: "애틱 타일 이지케어 발수러그",
          outside: true,
          pointX: 486,
          pointY: 197.667,
          priceOriginal: 76000,
          priceDiscount: 76000,
          discountRate: 0,
          imageUrl:
            "//cdn.ggumim.co.kr/cache/furniture/300/20220110174102naCtctXTxY.png",
        },
        {
          productId: 83544,
          productName: "플로윙 화이트 원형 테이블",
          outside: false,
          pointX: 319,
          pointY: 170.667,
          priceOriginal: 120000,
          priceDiscount: 74000,
          discountRate: 38,
          imageUrl:
            "//cdn.ggumim.co.kr/cache/furniture/300/20200916133257f6yBWCxVRR.jpg",
        },
        {
          productId: 134225,
          productName: "트리니 벽난로콘솔",
          outside: true,
          pointX: 288,
          pointY: 89.6666,
          priceOriginal: 99000,
          priceDiscount: 99000,
          discountRate: 0,
          imageUrl:
            "//cdn.ggumim.co.kr/cache/furniture/300/202104291640198PeKIa8W06.JPG",
        },
      ],
    },
  };

  test("matches snapshot", () => {
    const utils = render(<ProductImageContent {...initialProps} />);
    expect(utils.container).toMatchSnapshot();
  });
});
