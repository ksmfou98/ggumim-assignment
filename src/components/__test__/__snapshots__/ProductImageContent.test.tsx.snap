// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImageContent matches snapshot 1`] = `
<div>
  <div
    class="sc-ksdxgE"
  >
    <div
      class="sc-hBUSln eSbvKF"
    >
      <img
        alt="mainImage"
        class="sc-fotOHu kFHbar"
        src="//cdn.ggumim.co.kr/cache/star/1000/2022011017094316oRcWeb8R.jpeg"
      />
      <div
        class="sc-pVTFL bdhnqH"
      >
        <img
          alt="tag"
          class="sc-jrQzAO diZKiZ"
          src="tagIcon.png"
        />
      </div>
      <div
        class="sc-pVTFL bdhnqH"
      >
        <img
          alt="tag"
          class="sc-jrQzAO diZKiZ"
          src="tagIcon.png"
        />
      </div>
      <div
        class="sc-pVTFL bdhnqH"
      >
        <img
          alt="tag"
          class="sc-jrQzAO diZKiZ"
          src="tagIcon.png"
        />
      </div>
    </div>
    <div
      class="sc-egiyK licEs"
    >
      <div
        class="sc-bqiRlB bWAaSl"
      >
        <div
          class="sc-iqseJM aucdn"
        >
          <div
            class="sc-crHmcD bSca-dc"
          />
        </div>
        <div
          class="sc-iqseJM aucdn"
        >
          <div
            class="sc-crHmcD dAxqbr"
          >
            <div
              class="sc-kDTinF cBxmeF"
              data-testid="discountBadge"
            >
              38
              <span>
                %
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-iqseJM aucdn"
        >
          <div
            class="sc-crHmcD bWTLgX"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
